tests/PHPStan/Analyser/nsrt/bug-4700.php

```diff
@@ -40,7 +40,7 @@ function(array $array, int $count): void {
 	if (isset($array['d'])) $a[] = $array['d'];
 	if (isset($array['e'])) $a[] = $array['e'];
 	if (count($a) > $count) {
-		assertType('int<1, 5>', count($a));
+		assertType('int<2, 5>', count($a));
 		assertType('array{0: mixed~null, 1?: mixed~null, 2?: mixed~null, 3?: mixed~null, 4?: mixed~null}', $a);
```

Comment:
can/should this be more precise (separate PR)?

```suggestion
		assertType('array{0: mixed~null, 1: mixed~null, 2?: mixed~null, 3?: mixed~null, 4?: mixed~null}', $a);
```

Reply:
yes, it should. I would prefer separate, because I have no clue where to even look :)

apparently it was like this (same array and `int<2, 5>` before you added the count feature already. interesting that it changed then and changed back now.

Reply:
ok cool. I guess we should open a new issue for it to get issue-bot support and have it on the radar

Reply:
I could fix this in another commit. looks like the offset was not considered for the truthy scope. falsey can be improved too, but not really sure how, might need bigger changes.

I mean, if the array count is `int<2, 5>` in truthy, then it should most likely be `int<0, 1>` in falsey? I believe the `turnListIntoConstantArray()` helper might just not be able to deal with this yet.

Reply:
could simplify it a bit more and found a better fix. added comments to where the types could be better in falsey context, I think this is as far as I can go here. at least it's not wrong any more now, it could be better / narrower though :)